beacon-chain/core/transition/transition.go

```diff
@@ -223,50 +212,68 @@ func ProcessSlots(ctx context.Context, state state.BeaconState, slot primitives.
 		}
 		if cachedState != nil && !cachedState.IsNil() && cachedState.Slot() < slot {
 			highestSlot = cachedState.Slot()
-			state = cachedState
+			beaconState = cachedState
 		}
 	} else if err != nil {
 		return nil, err
 	}
 	defer func() {
 		SkipSlotCache.MarkNotInProgress(key)
 	}()
-
-	for state.Slot() < slot {
+	customErrFn := func(ctx context.Context, st state.BeaconState) error {
```

Comment:
reviewer please carefully look at the use of this function and let me know if this works well for our usecase